src/components/data/FilterBar.js

```diff
@@ -1,100 +1,97 @@
-import React from 'react';
-import {Field} from "../form/Field";
-import {FormModel} from "../../data/FormModel";
-import {DisplayModel} from "../../data/DisplayModel";
-import {FilterModel} from "../../data/FilterModel";
+import React from 'react'
+import { Field } from '../form/Field'
+import { FilterModel } from '../../data/FilterModel'
 
 export class FilterBar extends React.Component {
-    constructor(props) {
-        super(props);
+  constructor (props) {
+    super(props)
 
-        this.state = {
-            model: this.getModel(this.props.fields)
-        };
+    this.state = {
+      model: this.getModel(this.props.fields)
     }
+  }
 
-    componentDidMount() {
-        const model = this.getModel(this.props.fields);
-        this.setModel(model);
-    }
+  componentDidMount () {
+    const model = this.getModel(this.props.fields)
+    this.setModel(model)
+  }
 
-    componentDidUpdate(prevProps) {
-        if(this.props.fields !== prevProps.fields) {
-            const model = this.getModel(this.props.fields);
-            this.setModel(model);
-        }
+  componentDidUpdate (prevProps) {
+    if (this.props.fields !== prevProps.fields) {
+      const model = this.getModel(this.props.fields)
+      this.setModel(model)
     }
+  }
 
-    getModel = (fields = []) => {
-        return new FilterModel({fields});
-    };
+  getModel = (fields = []) => {
+    return new FilterModel({fields})
+  }
 
-    setModel(model) {
-        this.setState(() => ({
-            model
-        }));
-    }
+  setModel (model) {
+    this.setState(() => ({
+      model
+    }))
+  }
 
-    onFieldChange = (event) => {
-        const name = event.target.name;
-        const value = event.target.value;
-        const model = this.state.model;
-        model.set(name, value);
-        this.setModel(model);
-    };
+  onFieldChange = (event) => {
+    const name = event.target.name
+    const value = event.target.value
+    const model = this.state.model
+    model.set(name, value)
+    this.setModel(model)
+  }
 
-    onFilter = (event) => {
-        event.preventDefault();
-        const data = {};
-        this.state.model.fields.forEach((field) => {
-            if (field.value) {
-                data[field.name] = field.value
-            }
-        });
+  onFilter = (event) => {
+    event.preventDefault()
+    const data = {}
+    this.state.model.fields.forEach((field) => {
+      if (field.value) {
+        data[field.name] = field.value
+      }
+    })
 
-        this.props.onFilter(data);
-    };
+    this.props.onFilter(data, true)
```

Comment:
What is the purpose of the `true` parameter here?

Reply:
This parameter tells the onFilter function whether or not to reset the filter params (as in remove the existing filter params). 
[Here's the actual function](https://github.com/flickerleap/react-admin-template/blob/2e779c4e16d4ffc8b2faf0b85b8e35b31dcace01/src/components/screens/IndexScreen.js#L93)
[Here's the function that the param affects](https://github.com/flickerleap/react-admin-template/blob/2e779c4e16d4ffc8b2faf0b85b8e35b31dcace01/src/components/screens/IndexScreen.js#L62)